doc/source/security/encryption-functions.rst

```diff
@@ -0,0 +1,438 @@
+
+.. rubric:: Returns
+
+The digest of the given string as a binary string
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* digest_type - the supported values are the following (based on the OpenSSL version):
+
+  .. list-table::
+      :header-rows: 1
+
+      * - Value Name for OpenSSL 1.0.2
+        - Value Name for OpenSSL 1.1.x addition
+      * - 'MD5'
+        - 'BLAKE2B512'
+      * - 'SHA1'
+        - 'BLAKE2S256'
+      * - 'SHA224'
+        - 'RIPEMD'
+      * - 'SHA256'
+        - 'RMD160'
+      * - 'SHA384'
+        - 'SHAKE128'
+      * - 'SHA512'
+        - 'SHAKE256'
+      * - 'MD4'
+        - 'SM3'
+      * - 'RIPEMD160'
+        - 'WHIRLPOOL'
+
+* str - String used to generate the digest string.
+
+.. _encryption_variables:
+
+.. rubric:: Encryption threshold variables
+
+The maximum key length limits are defined by OpenSSL. Server administrators can limit the maximum key length using the encryption threshold variables. 
+
+The variables are automatically registered when `component_encryption_udf` is installed.
+
+
+.. list-table::
+   :header-rows: 1
+
+   * - Variable Name
+   * - :ref:`dh_threshold`
+
+.. _dh_threshold:
+
+.. rubric:: ``encryption_udf.dh_bits_threshold``
+
+These variables set the threshold limits for the :ref:`create_dh_param`. The value takes precedence over the OpenSSL maximum length value.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+   * - default
+     - 10000
+
+The range for this variable is from 1024 to 10,000. The default value is 10,000.
+
+.. _dsa_threshold:
+
+.. rubric:: `encryption_udf.dsa_bits_threshold`
+
+These variables set the threshold limits for :ref:`asymmetric_encrypt <asym_encrypt>`. The value takes precedence over the OpenSSL maximum length value.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+   * - default
+     - 9984
+
+
+The range for this variable is from 1,024 to 9,984. The default value is 9,984.
+
+.. _rsa_threshold:
+
+.. rubric:: `encryption_udf.rsa_bits_threshold`
+
+These variable sets the threshold limits for the :ref:`asymmetric_encrypt <asym_encrypt>`. The value takes precedence over the OpenSSL maximum length value.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+   * - default
+     - 16384
+
+The range for this variable is from 1,024 to 16,384.
+
+.. _function_examples:
+
+.. rubric:: Examples
+
+The following example sets the threshold variables, creates a private key, creates a public key, encrypts data, and decrypts data.
+
+
+.. code-block:: mysql
+
+    -- Set Global variable
+    mysql> SET GLOBAL encryption_udf.dh_bits_threshold = 4096;
```

Comment:
This example is slightly incorrect.
Change the SQL to
SET GLOBAL encryption_udf.rsa_bits_threshold = 4096;

Reply:
added